src/code/z_vimode.c

```diff
@@ -1,23 +1,346 @@
+            },
+            {
+                // OS_TV_MPAL
+                BURST(57, 30, 5, 70),
+                VSYNC(524),
+                HSYNC(3088, 0),
+                LEAP(3100, 3100),
+                HSTART(108, 748),
+                START(37, 511),
+                BURST(4, 2, 14, 0),
+            },
+        };
+        const ViModeStruct* ptr = &D_801DF0C0[tvType];
+
+        viMode->comRegs.burst = ptr->unk_00;
+        viMode->comRegs.vSync = ptr->unk_04;
+        viMode->comRegs.hSync = ptr->unk_08;
+        viMode->comRegs.leap = ptr->unk_0C;
+        viMode->comRegs.hStart = ptr->unk_10;
+        viMode->fldRegs[0].vStart = ptr->unk_14;
+        viMode->fldRegs[1].vStart = ptr->unk_14;
+        viMode->fldRegs[0].vBurst = ptr->unk_18;
+        viMode->fldRegs[1].vBurst = ptr->unk_18;
+    } else {
+        __assert("../z_vimode.c", 216);
+    }
+
+    viMode->comRegs.hStart += (leftAdjust << 16) + (s16)rightAdjust;
+    viMode->fldRegs[0].vStart += (upperAdjust << 16) + (s16)lowerAdjust;
+    viMode->fldRegs[1].vStart += (upperAdjust << 16) + (s16)lowerAdjust;
+
+    if (loResDeinterlaced) {
+        viMode->comRegs.vSync++;
+        if (tvType == OS_TV_MPAL) {
+            viMode->comRegs.hSync += HSYNC(1, 4);
+            viMode->comRegs.leap += LEAP((u16)-4, (u16)-2);
+        }
+    } else {
+        viMode->fldRegs[0].vStart += START((u16)-3, (u16)-2);
+        if (tvType == OS_TV_MPAL) {
+            viMode->fldRegs[0].vBurst += BURST((u8)-2, (u8)-1, 12, -1);
+        } else if (tvType == OS_TV_PAL) {
+            viMode->fldRegs[1].vBurst += BURST((u8)-2, (u8)-1, 2, 0);
+        }
+    }
+
+    viMode->comRegs.xScale = (SCREEN_WIDTH << 10) / SCREEN_WIDTH;
+    viMode->comRegs.vCurrent = 0;
+
+    viMode->fldRegs[0].origin = ORIGIN(width * 2 * (fb16Bit ? 1 : 2));
+    viMode->fldRegs[1].origin = ORIGIN(width * 2 * (fb16Bit ? 1 : 2) * (loRes ? 1 : 2));
+
+    viMode->fldRegs[0].yScale = yScaleLo | yScaleHiEvenField;
+    viMode->fldRegs[1].yScale = yScaleLo | yScaleHiOddField;
+
+    viMode->fldRegs[0].vIntr = 2;
+    viMode->fldRegs[1].vIntr = 2;
+}
+
+void ViMode_Save(ViMode* viMode) {
+    R_VI_MODE_EDIT_STATE = viMode->editState;
+    R_VI_MODE_EDIT_WIDTH = viMode->viWidth;
+    R_VI_MODE_EDIT_HEIGHT = viMode->viHeight;
+    R_VI_MODE_EDIT_ULY_ADJ = viMode->upperAdjust;
+    R_VI_MODE_EDIT_LRY_ADJ = viMode->lowerAdjust;
+    R_VI_MODE_EDIT_ULX_ADJ = viMode->leftAdjust;
+    R_VI_MODE_EDIT_LRX_ADJ = viMode->rightAdjust;
+
+    if (SREG(58) == 1) {
+        SREG(58) = 0;
+        switch (SREG(59)) {
+            case 1:
+                ViMode_LogPrint(&osViModePalLan1);
+                break;
+            case 2:
+                ViMode_LogPrint(&osViModeFpalLan1);
+                break;
+            default:
+                ViMode_LogPrint(&viMode->customViMode);
+                break;
+        }
+    }
+}
+
+void ViMode_Load(ViMode* viMode) {
+    if ((R_VI_MODE_EDIT_WIDTH & ~3) == 1) {
```

Comment:
```suggestion
void ViMode_Load(ViMode* viMode) {
    //! @bug This condition never passes as the lowest bit is always masked out to 0
    if ((R_VI_MODE_EDIT_WIDTH & ~3) == 1) {
```
Seems like this bug carried over to MM